src/main/java/TaskList.java

```diff
@@ -0,0 +1,135 @@
+                try {
+                    String[] parseDateTime = dateTime.split(" ",  2);
+                    LocalDate localDate = LocalDate.parse(parseDateTime[0]);
+                    if (parseDateTime.length > 1) {
+                        String time = parseDateTime[1];
+                        newTask = new Deadline(task, preposition, localDate, time);
+                    } else {
+                        newTask = new Deadline(task, preposition, localDate);
+                    }
+                } catch (DateTimeParseException e) {
+                    System.err.println("Error parsing date and time. Please input date and time as YYYY-mm-dd hh:mm");
+                }
+
+            } else { //case when taskType is "event"
+                String[] in = taskDesc.split("/", 2);
+                String task = in[0].trim();
+                String dateAndPreposition = in[1];
+                String[] res = dateAndPreposition.split(" ", 2);
+                String preposition = res[0];
+                String dateTime = res[1];
+                try {
+                    String[] parseDateTime = dateTime.split(" ",  2);
+                    LocalDate localDate = LocalDate.parse(parseDateTime[0]);
+                    if (parseDateTime.length > 1) {
+                        String time = parseDateTime[1];
+                        newTask = new Event(task, preposition, localDate, time);
+                    } else {
+                        newTask = new Event(task, preposition, localDate);
+                    }
+                } catch (DateTimeParseException e) {
+                    System.err.println("Error parsing date and time. Please input date and time as YYYY-mm-dd hh:mm");
+                }
+            }
+
+            if (Objects.isNull(newTask)) {
+                System.err.println("Attempting to add invalid task. Operation aborted.");
+            } else {
+                taskList.add(newTask);
+                System.out.println("     Got it. I've added this task:");
+                System.out.println("       " + newTask);
+                System.out.println("     Now you have " + taskList.size() + " tasks in the list.");
+            }
+        } catch (IndexOutOfBoundsException e) {
+            System.err.println("     ☹ OOPS!!! The description of a task cannot be empty.");
+        } finally {
+            storage.saveTask(taskList);
+        }
+    }
+
+    /**
+     * Deletes task of taskName from taskList.
+     * @param taskName Task to be deleted
+     */
+    public void deleteTask(String taskName) {
+        try {
+            String taskNum = taskName.split(" ", 2)[1];
+            Task currTask = taskList.get(Integer.parseInt(taskNum) - 1);
+            taskList.remove(currTask);
+            System.out.println("     Noted. I've removed this task: ");
+            System.out.println("       " + currTask);
+            System.out.println("     Now you have " + taskList.size() + " tasks in the list.");
```

Comment:
You can consider letting the UI class handle all the printing e.g. ui.showDelete(task)